To: vim_dev@googlegroups.com
Subject: Patch 7.3.193
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.193
Problem:    In the command line window ":close" doesn't work properly. (Tony
	    Mechelynck)
Solution:   Use Ctrl_C instead of K_IGNORE for cmdwin_result. (Jean-Rene
	    David)
Files:	    src/ex_docmd.c, src/ex_getln.c


*** ../vim-7.3.192/src/ex_docmd.c	2011-05-10 16:41:13.000000000 +0200
--- src/ex_docmd.c	2011-05-19 14:42:56.000000000 +0200
***************
*** 6472,6478 ****
  {
  # ifdef FEAT_CMDWIN
      if (cmdwin_type != 0)
! 	cmdwin_result = K_IGNORE;
      else
  # endif
  	if (!text_locked()
--- 6472,6478 ----
  {
  # ifdef FEAT_CMDWIN
      if (cmdwin_type != 0)
! 	cmdwin_result = Ctrl_C;
      else
  # endif
  	if (!text_locked()
*** ../vim-7.3.192/src/ex_getln.c	2011-04-28 17:21:49.000000000 +0200
--- src/ex_getln.c	2011-05-19 14:42:56.000000000 +0200
***************
*** 6324,6329 ****
--- 6324,6335 ----
  	    ccline.cmdbuff = vim_strsave((char_u *)"qa");
  	    cmdwin_result = CAR;
  	}
+ 	else if (cmdwin_result == Ctrl_C)
+ 	{
+ 	    /* :q or :close, don't execute any command
+ 	     * and don't modify the cmd window. */
+ 	    ccline.cmdbuff = NULL;
+ 	}
  	else
  	    ccline.cmdbuff = vim_strsave(ml_get_curline());
  	if (ccline.cmdbuff == NULL)
*** ../vim-7.3.192/src/version.c	2011-05-19 14:30:07.000000000 +0200
--- src/version.c	2011-05-19 14:48:12.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     193,
  /**/

-- 
There's no place like $(HOME)!

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
